process/block/preprocess/transactions.go

```diff
@@ -1076,6 +1074,7 @@ func (txs *transactions) ProcessMiniBlock(miniBlock *block.MiniBlock, haveTime f
 			return processedTxHashes, err
 		}
 
+		processedTxHashes = append(processedTxHashes, miniBlockTxHashes[index])
```

Comment:
You should move this line back, and use another variable for your scope, as if it would be an error in ProcessTransaction at any point, then on defer, the gas consumed would be reverted only for the txs which has been  processed and for the others the gas would remain not reverted.  Actually computeGasConsumed in this point has been executed for all of them.

Reply:
done.